scripts/build-toolchains.sh

```diff
@@ -8,31 +10,149 @@ unamestr=$(uname)
 RDIR=$(pwd)
 : ${CHIPYARD_DIR:=$(pwd)} #default value is the PWD unless overridden
 
-if [ $# -ne 0 ]; then
-  TOOLCHAIN=$1
-  if [ $1 == "riscv" ]; then
-    TOOLCHAIN="riscv-tools"
-  elif [ $1 == "hwacha" ]; then
-    TOOLCHAIN="esp-tools"
-  fi
-else
-  TOOLCHAIN="riscv-tools"
+PRECOMPILED_REPO_HASH=56a40961c98db5e8f904f15dc6efd0870bfefd9e
+
+function usage
+{
+    echo "usage: ./scripts/build-toolchains.sh [riscv] [hwacha] [ ec2fast | --ec2fast] "
+    echo "   riscv: if set, builds the riscv toolchain (this is also the default)"
+    echo "   hwacha: if set, builds esp-tools toolchain"
+    echo "   ec2fast: if set, pulls in a pre-compiled RISC-V toolchain for an EC2 manager instance"
+}
+
+#taken from riscv-tools to check for open-ocd autoconf versions
+check_version() {
+    $1 --version | awk "NR==1 {if (\$NF>$2) {exit 0} exit 1}" || (
+        echo $3 requires at least version $2 of $1. Aborting.
+        exit 1
+    )
+}
+
+if [ "$1" == "--help" -o "$1" == "-h" -o "$1" == "-H" ]; then
+    usage
+    exit 3
+fi
+
+TOOLCHAIN="riscv-tools"
+EC2FASTINSTALL="false"
+FASTINSTALL="false"
+while test $# -gt 0
+do
+   case "$1" in
+        riscv)
+            TOOLCHAIN="riscv-tools"
+            ;;
+        hwacha)
+            TOOLCHAIN="esp-tools"
+            ;;
+        ec2fast | --ec2fast) # I don't want to break this api
+            EC2FASTINSTALL=true
+            ;;
+        -h | -H | --help)
+            usage
+            exit
```

Comment:
If we're gonna use weird exit values at least make them consistent for the same failure case. I.e. this should be `exit 3` based on the if statement before this while.

Reply:
done